blocks/identity-block/components/SocialSignOn/index.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { isServerSide } from '@wpmedia/engine-theme-sdk';
+import PropTypes from '@arc-fusion/prop-types';
+import useIdentity from '../Identity';
+import './styles.scss';
+
+const SocialSignOn = ({ onError, redirectURL }) => {
+  const { Identity } = useIdentity();
+  const [config, setConfig] = useState(() => Identity?.configOptions ?? {});
+  const [isGoogleInitialized, setIsGoogleInitialized] = useState(false);
+  const [isFacebookInitialized, setIsFacebookInitialized] = useState(false);
+
+  const isGoogleReset = useRef(false);
+
+  if (!isServerSide() && !window.onFacebookSignOn) {
+    window.onFacebookSignOn = async () => {
+      try {
+        await Identity.facebookSignOn();
+        window.location = redirectURL;
+      } catch (e) {
+        onError();
+      }
+    };
+  }
+
+  useEffect(() => {
+    const fetchConfig = async () => {
+      await Identity.getConfig();
+      setConfig(Identity.configOptions);
+    };
+    if (Identity && !Identity.configOptions) {
+      fetchConfig();
+    }
+  }, [Identity]);
+
+  useEffect(() => {
+    const initializeFacebook = async () => {
+      await Identity.initFacebookLogin(null);
+    };
+    const initializeGoogle = async () => {
+      await Identity.initGoogleLogin(null, {
+        width: 300,
+        height: 48,
+        onSuccess: () => {
+          if (!isGoogleReset.current) {
+            isGoogleReset.current = true;
+          } else {
+            window.location = redirectURL;
+          }
+        },
+        onFailure: () => { onError(); },
+      });
+    };
+    if (config.googleClientId && !isGoogleInitialized) {
+      initializeGoogle();
+      setIsGoogleInitialized(true);
+    }
+    if (config.facebookAppId && !isFacebookInitialized) {
+      initializeFacebook();
+      setIsFacebookInitialized(true);
+    }
+  }, [Identity,
+    config.facebookAppId,
+    config.googleClientId,
+    isFacebookInitialized,
+    isGoogleInitialized,
+    onError,
+    redirectURL]);
+
+  if (!Identity) {
+    return null;
+  }
+
+  if (!config.facebookAppId && !config.googleClientId) {
+    return null;
```

Comment:
throw error?

Reply:
I suppose if we had some sort of admin messaging that said they likely haven't configured their social media keys, but otherwise, I think this is OK.  There's nothing to do if the IDs don't exist.  It's not great from the admin perspective, but what is?